### PR TITLE
Add mirror and default ForceFormat for OutputPipeArgument

### PR DESCRIPTION
**Pull request by GorobVictor**

### CODE REVIEW

File: FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace FFMpegCore.Arguments
+{
+    public class SetMirrorVideo : IArgument
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs]: Should not be a regular IArgument but instead a VideoFilter argument (see `ScaleArgument` for example of this)

File: FFMpegCore/FFMpeg/FFMpegArguments.cs [hunk 2]
```diff
@@ -57,6 +57,12 @@ private FFMpegArgumentProcessor ToProcessor(IOutputArgument argument, Action<FFM
         {
             var args = new FFMpegArgumentOptions();
             addArguments?.Invoke(args);
+
+            if (argument is OutputPipeArgument && !Arguments.Any(x => x is ForceFormatArgument))
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/FFMpegArguments.cs]: I think it may make more sense to include format directly in the `.OutputToPipe(..)` call than this. 
Secondly, I'd prefer separate features as separate PRs, though I won't block this PR for that reason

**Review by rosenbjerg — CHANGES_REQUESTED** [review]

File: FFMpegCore/FFMpeg/FFMpegArguments.cs (hunk 2 above)
Comment (GorobVictor) [reply FFMpegCore/FFMpeg/FFMpegArguments.cs]: it is better to leave it as it is for better code readability

File: FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs (hunk 1 above)
Comment (GorobVictor) [reply FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs]: I little remade. But left IArgument, since flip does not have a key and therefore IVideoFilter does not fit.

https://duxyng.wordpress.com/2013/04/07/rotateflip-video-with-ffmpeg/

Comment (rosenbjerg) [reply FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs]: Any video filter (-vf) argument MUST be a `IVideoFilterArgument` and if you want it added, you should also provide a method for the fluent argument builder.
Again, see [ScaleArgument](https://github.com/rosenbjerg/FFMpegCore/blob/master/FFMpegCore/FFMpeg/Arguments/ScaleArgument.cs)
or 
[TransposeArgument](https://github.com/rosenbjerg/FFMpegCore/blob/master/FFMpegCore/FFMpeg/Arguments/TransposeArgument.cs)

File: FFMpegCore/FFMpeg/Enums/Mirror.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace FFMpegCore.Enums
+{
+    public enum Mirror : byte
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Enums/Mirror.cs]: Please rename to `VideoFilterMirroring` and why the need to inherit from byte?

File: FFMpegCore/FFMpeg/Enums/Mirror.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    {
+        Verticall = 1,
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Enums/Mirror.cs]: Vertical

File: FFMpegCore/FFMpeg/Enums/Mirror.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+        Horizontall = 2
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Enums/Mirror.cs]: Horizontal

File: FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs [hunk 6]
```diff
@@ -1,11 +1,34 @@
...
+            get
+            {
+                switch (_value)
+                {
+                    case Mirror.Horizontall:
+                        return "-vf \"hflip\"";
+                    case Mirror.Verticall:
+                        return "-vf \"vflip\"";
+                    default:
+                        throw new Exception("SetMirrorVideo: argument not found");
```
Comment (rosenbjerg) [inline FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs]: Should be an ArgumentOutOfRangeException

File: FFMpegCore/FFMpeg/FFMpegArguments.cs (hunk 2 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMpeg/FFMpegArguments.cs]: I do not like this feature of automagically adding a forceformatargument without the user's knowledge.

File: FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs (hunk 1 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMpeg/Arguments/SetMirrorVideo.cs]: It needs to be a `IVideoFilterArgument` since ffmpeg only accepts a single -vf.
Just set the Key to vflip/hflip and leave the value as an empty string and I'll make that work when merging

File: FFMpegCore/FFMpeg/FFMpegArguments.cs (hunk 2 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMpeg/FFMpegArguments.cs]: And what improves code readability?
